sim/pokemon.ts

```diff
@@ -1776,7 +1776,8 @@ export class Pokemon {
 		}
 		this.ability = ability.id;
 		this.abilityState = {id: ability.id, target: this};
-		if (ability.id && this.battle.gen > 3 && !(oldAbility === ability.id && this.battle.gen > 4)) {
+		if (ability.id && this.battle.gen > 3 &&
+			!(!isFromFormeChange && oldAbility === ability.id && this.battle.gen > 4)) {
```

Comment:
```suggestion
			(isFromFormeChange || oldAbility !== ability.id || this.battle.gen <= 4)) {
```
Simplified by distributing out the NOT